parsers/numword.go

```diff
@@ -0,0 +1,94 @@
+// Given a number in string form
+// Return the number in word form of the greatest power (e.g. 300,100,000,000 => 300.1 Billion)
+// Lower limit > Thousands (e.g. xxx,000) # Numbers below this are in hundreds and can be expressed numerically
+// Upper limit > Centillion (10^303) ref: https://en.wikipedia.org/wiki/Names_of_large_numbers#Standard_dictionary_numbers
+
+package parsers
+
+import (
+  "math"
+  "regexp"
+  "strings"
+  "strconv"
+)
+
+// NumberWord handles strings made of contiguous "0-9" characters
+// converts to word strings of the greatest power
+type NumberWord struct{}
+
+// NewNumberWord construcs a NumberWord struct
+func NewNumberWord() *NumberWord {
+  return &NumberWord{}
+}
+
+// CanParseIntoHuman ...
+func (n *NumberWord) CanParseIntoHuman(s string) bool {
+  // is it 4 or more characters? (e.g. is it => 1000)
+  if len(s) < 4 {
+    return false
+  }
+
+  // is it a (delimited[,. ]) number?
+  match, _ := regexp.MatchString(`^(([0-9]+)|([0-9]{1,3}[., ])+[0-9]{1,3})$`, s)
+  if match {
+    // is it less than the max? 
+      //  300000 max places plus delimiters
+    if len(s) < 400000 {
```

Comment:
Since this is the only condition being used once we're inside the "we have regex match" branch, we can combine these branches into 1 ie:

```golang
if match && len(s) < 400000 {
   ...
```